vsgenstubs4/_vapoursynth.part.pyi

```diff
@@ -516,236 +843,327 @@ class VideoFormat:
     subsampling_h: int
     num_planes: int
 
-    def __int__(self) -> int: ...
+    def __init__(self) -> NoReturn: ...  # type: ignore[misc]
 
-    def _as_dict(self) -> typing.Dict[str, typing.Any]: ...
-    def replace(self, *,
-                color_family: typing.Optional[ColorFamily] = None,
-                sample_type: typing.Optional[SampleType] = None,
-                bits_per_sample: typing.Optional[int] = None,
-                subsampling_w: typing.Optional[int] = None,
-                subsampling_h: typing.Optional[int] = None
-                ) -> 'VideoFormat': ...
+    def _as_dict(self) -> _VideoFormatInfo: ...
 
+    def replace(
+        self, *,
+        color_family: Union[ColorFamily, None] = None,
+        sample_type: Union[SampleType, None] = None,
+        bits_per_sample: Union[int, None] = None,
+        subsampling_w: Union[int, None] = None,
+        subsampling_h: Union[int, None] = None
+    ) -> 'VideoFormat': ...
+
+    @overload
+    def __eq__(self, other: 'VideoFormat') -> bool: ...  # type: ignore[misc]
+
+    @overload
+    def __eq__(self, other: Any) -> Literal[False]: ...
 
-_FramePropsValue = typing.Union[
-    SingleAndSequence[int],
-    SingleAndSequence[float],
-    SingleAndSequence[str],
-    SingleAndSequence['VideoNode'],
-    SingleAndSequence['VideoFrame'],
-    SingleAndSequence['AudioNode'],
-    SingleAndSequence['AudioFrame'],
-    SingleAndSequence[typing.Callable[..., typing.Any]]
-]
 
-class FrameProps(typing.MutableMapping[str, _FramePropsValue]):
+class FrameProps(MutableMapping[str, _VapourSynthMapValue]):
+    def __init__(self) -> NoReturn: ...  # type: ignore[misc]
 
-    def copy(self) -> typing.Dict[str, _FramePropsValue]: ...
+    def setdefault(
+        self, key: str, default: _VapourSynthMapValue = 0
+    ) -> _VapourSynthMapValue: ...
 
-    def __getattr__(self, name: str) -> _FramePropsValue: ...
-    def __setattr__(self, name: str, value: _FramePropsValue) -> None: ...
+    def copy(self) -> 'FrameProps': ...
+
+    def __setattr__(self, name: str, value: _VapourSynthMapValue) -> None: ...
+
+    def __getattr__(self, name: str) -> _VapourSynthMapValue: ...
+
+    def __delattr__(self, name: str) -> None: ...
+
+    def __setitem__(self, name: str, value: _VapourSynthMapValue) -> None: ...
+
+    def __getitem__(self, name: str) -> _VapourSynthMapValue: ...
 
-    # mypy lo vult.
-    # In all seriousness, why do I need to manually define them in a typestub?
```

Comment:
These two lines describe why these dunder methods exist.
Please check if they are still required, and if not, remove these dunder methods at all.
If they are required, please provide a suitable comment on the rewritten .pyi file.